_posts/2021-04-07-upgrade-improvements.md

```diff
@@ -0,0 +1,71 @@
+
+Here's how Strimzi handles certain situations:
+* Supported Kafka `version` and `log.message.format.version` or `inter.broker.protocol.version` is set in CR - Strimzi will keep the values and missing one will se based on `version`.
+* Supported Kafka `version` is set, `log.message.format.version` and `inter.broker.protocol.version` are not set in CR - Strimzi will set `log.message.format.version` and `inter.broker.protocol.version` based on `version`
+* Supported Kafka `version` is not set in CR - Strimzi will set Kafka `version`, `log.message.format.version` and `inter.broker.protocol.version` based on the default Kafka supported by Strimzi.
+* Unsupported Kafka `version` is set in CR - Strimzi will set `Kafka` CR status to `NotReady`. 
+  In that case user has to update CR and manually set a proper config.
+
+You don't need to configure the Kafka `version`, `log.message.format.version` or `inter.broker.protocol.version` before the upgrade, but we **strongly** recommend it.
+
+#### Upgrade without versions
+
+Strimzi is able to deal with upgrade of Kafka clusters where versions are not set in the custom resource.
+In cases like this, Strimzi determines the versions and performs the upgrade as follows:
+* Get the version of the currently deployed Kafka and set it in the custom resource.
+* Perform a rolling update of the Kafka cluster to change the images from the old Strimzi version to the new one.
+* Upgrade the Kafka `version` to the default version supported by the new Strimzi version, which will perform a rolling update.
+* Get `log.message.format.version` and `inter.broker.protocol.version` based on `version`.
+* Set `inter.broker.protocol.version`, and perform a rolling update of the Kafka cluster.
+* Set `log.message.format.version`, and perform a rolling update of the Kafka cluster.
+
+You can see the states of the upgrade process and each rolling update in the diagram below.
+_Note that Kafka versions are only for an illustration.
+Real Kafka versions should be set based on Strimzi supported ones._
+
+![Kafka upgrade states](/assets/images/posts/2021-04-19-kafka-rolling-updates.png)
+
+#### Things to be aware of
+
+With the introduction of these upgrade improvements, there are a couple of things to be aware of:
+* For upgrade, you need to roll out the new brokers while first using the older `log.message.format.version` or `inter.broker.protocol.version`, and only afterwards change to the new versions for message format and inter-broker protocol.
+* [Downgrade is supported](https://strimzi.io/docs/operators/0.24.0/full/deploying.html#con-target-downgrade-version-str) only when you don't set new versions of `log.message.format.version` and `inter.broker.protocol.version`. 
+For instance, if you upgrade Strimzi to `X+1` from `X`, and upgrade Kafka to `y.x.w`, a working downgrade is not guaranteed after you set `log.message.format.version` or `inter.broker.protocol.version` to `y.x`.
```

Comment:
Where Strimzi X doesn't support y.x but X+1 does.

Reply:
I updated the line. Now there is `y.z.w` and `y.z-1`